mrp_subcontracting_lot_number_propagation/models/stock_move_line.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from lxml import etree
+
+from odoo import api, fields, models
+from odoo.osv import expression
+from odoo.tools.safe_eval import safe_eval
+
+from odoo.addons.base.models.ir_ui_view import (
+    transfer_modifiers_to_node,
+    transfer_node_to_modifiers,
+)
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    is_lot_id_editable = fields.Boolean(
+        string="Is lot editable?",
+        compute="_compute_is_lot_id_editable",
+        help="Technical field to define if the 'lot_id' field is editable.",
+    )
+
+    @api.depends("lot_id")
+    def _compute_is_lot_id_editable(self):
+        for line in self:
+            line.is_lot_id_editable = True
+            mo_ids = line.move_id.move_orig_ids.mapped("production_id")
```

Comment:
So we know it's not a list of IDs but a recordset
```suggestion
            mos = line.move_id.move_orig_ids.mapped("production_id")
```